endpoints/openrtb2/auction_test.go

```diff
@@ -5618,6 +5618,50 @@ func TestValidateOrFillCookieDeprecation(t *testing.T) {
 	}
 }
 
+func TestSetGPCImplicitly(t *testing.T) {
+	testCases := []struct {
+		description string
+		headerValue string
+		expectedGPC string
+		expectError bool
+	}{
+		{
+			description: "Sec-GPC header set to '1', GPC should be set to '1'",
+			headerValue: "1",
+			expectedGPC: "1",
+			expectError: false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.description, func(t *testing.T) {
+			httpReq := &http.Request{
+				Header: http.Header{
+					"Sec-GPC": []string{test.headerValue},
+				},
+			}
+
+			r := &openrtb_ext.RequestWrapper{
+				BidRequest: &openrtb2.BidRequest{}, // what I should add here???
+			}
+
+			err := setGPCImplicitly(httpReq, r)
+			if (err != nil) != test.expectError {
+				t.Errorf("Unexpected error state: got %v, expected error: %v", err, test.expectError)
+			}
+
+			regExt, _ := r.GetRegExt()
+			gpcValue := regExt.GetGPC()
+
+			if gpcValue != nil && *gpcValue != test.expectedGPC {
+				t.Errorf("Expected GPC %v, but got %v", test.expectedGPC, *gpcValue)
+			} else if gpcValue == nil && test.expectedGPC != "" {
+				t.Errorf("Expected GPC to be set, but it was nil")
+			}
```

Comment:
Use the testify package to verify:
```
assert.NoError(t, r.RebuildRequest())
assert.JSONEq(t, string(test.expectedRegs.Ext), string(r.Regs.Ext))
```

You'll see here that I'm calling `RebuildRequest` so that we can be sure that any change to the regs ext struct on the wrapper is accurately reflected in the raw ext on the bid request. It also helps cut down the amount of test code you need here as you don't need to call `GetRegExt` and `GetGPC` etc.

Note that this `JSONEq` statement fails. I know why 😄. I am leaving it to you to try to figure it out as it will be a good exercise to learn how to debug your code/tests.